internal/factsengine/gatherers/dir_scan_test.go

```diff
@@ -0,0 +1,154 @@
+package gatherers_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+const dirScanTestBasePath = "/var/test"
+const dirScanTestGlobPattern = "/var/test/*/ASCS*"
+
+type DirScanGathererSuite struct {
+	suite.Suite
+	testFS afero.Fs
+}
+
+func TestDirScanGathererSuite(t *testing.T) {
+	suite.Run(t, new(DirScanGathererSuite))
+}
+
+func (s *DirScanGathererSuite) SetupSuite() {
+	tFs := afero.NewMemMapFs()
+
+	for i := 0; i <= 2; i++ {
```

Comment:
Only for 3 entries, it would be even simpler to have them in the same way you create the "bad" files hehe
It is easier to visually see the created folders